indexer/osm_editor.hpp

```diff
@@ -0,0 +1,126 @@
+#pragma once
+
+#include "geometry/rect2d.hpp"
+
+#include "indexer/feature_meta.hpp"
+#include "indexer/mwm_set.hpp"
+
+#include "editor/xml_feature.hpp"
+
+#include "base/timer.hpp"
+
+#include "std/ctime.hpp"
+#include "std/function.hpp"
+#include "std/map.hpp"
+#include "std/set.hpp"
+#include "std/string.hpp"
+
+struct FeatureID;
+class FeatureType;
+class Index;
+
+namespace osm
+{
+class Editor final
+{
+  Editor() = default;
+
+public:
+  using TMwmIdByMapNameFn = function<MwmSet::MwmId(string const & /*map*/)>;
+  using TInvalidateFn = function<void()>;
+  using TFeatureLoaderFn = function<unique_ptr<FeatureType> (FeatureID const & /*fid*/)>;
+
+  enum class FeatureStatus
+  {
+    Untouched,
+    Deleted,
+    Modified,
+    Created
+  };
+
+  static Editor & Instance();
+
+  void SetMwmIdByNameAndVersionFn(TMwmIdByMapNameFn const & fn) { m_mwmIdByMapNameFn = fn; }
+  void SetInvalidateFn(TInvalidateFn const & fn) { m_invalidateFn = fn; }
+  void SetFeatureLoaderFn(TFeatureLoaderFn const & fn) { m_featureLoaderFn = fn; }
+
+  void LoadMapEdits();
+
+  using TFeatureIDFunctor = function<void(FeatureID const &)>;
```

Comment:
Please, move these usings to the top of the public section, near other using declarations.